plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -0,0 +1,291 @@
+        crossing-island is formed.  That is, if ``threshold=5`` then consecutive
+        crossing-points are considered to be in the same crossing-island if
+        they are within 5 index steps of each other. (Default: 1)
+
+    min_points: positive `int` or `float`
+        Minimum number of data points required for the fitting to be applied to.
+        See **How it works** above for additional details.  The following list
+        specifies the optional values:
+
+        - ``min_points = None`` (Default) The largest of 5 and
+          ``factor * array_size`` is taken, where ``array_size`` is the size of
+          ``voltage`` and ``factor = 0.1`` for ``fit_type = "linear"`` and
+          ``0.2`` for ``"exponential"``.
+        - ``min_points = 0`` The entire passed array is fitted.
+        - ``min_points >= 1`` Exact minimum number of points.
+        - ``0 < min_points < 0`` The minimum number of points is taken as
+          ``min_points * array_size``.
+
+    fit_type: str
+        The type of curve to be fitted to the Langmuir trace,  ``"linear"`` or
+        ``"exponential"`` (Default).  This selects which `FitFunction` class
+        should be applied to the trace.
+
+        +-------------+----------------------------------------------------------+
+        | linear      | `~plasmapy.analysis.fit_functions.Linear`                |
+        +-------------+----------------------------------------------------------+
+        | exponential | `~plasmapy.analysis.fit_functions.ExponentialPlusOffset` |
+        +-------------+----------------------------------------------------------+
+
+    Returns
+    -------
+    vf: `float` or `numpy.nan`
+        The calculated floating potential (same units as the `voltage` array).
+        Returns `numpy.nan` if the floating potential can not be determined.
+        How :math:`V_f` is calculated depends on the fit function.  This is
+        described in the `root_solve()` method of the relevant fit function
+        (e.g. the
+        :meth:`~plasmapy.analysis.fit_functions.ExponentialPlusOffset.root_solve`
+        method of `~plasmapy.analysis.fit_functions.ExponentialPlusOffset`).
+
+    vf_err: `float` or `numpy.nan`
+        The uncertainty associated with the floating potential calculation
+        (units same as ``vf``).  Returns `numpy.nan` if the floating potential
+        can not be determined.  Like :math:`V_f`:, the calculation depends on
+        the applied fit function.  The `root_solve()` method also describes
+        how this is calculated.
+
+    rsq: `float`
+        The coefficient of determination (r-squared) value of the fit.  See the
+        documentation of the ``rsq`` property on the associated fit function
+        (e.g. the
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset.rsq`
+        property of
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset`).
+
+    func: sub-class of `~plasmapy.analysis.fit_functions.AbstractFitFunction`
```

Comment:
I still prefer `func`, because I don't think end users will be using the full functionality of the fit functions here.  The returned fit function is callable so users will typically just call the "function" to plot against their data.  So I am kinda hiding the complex machinery behind the scenes, but that's intentional.